Tools/2MGFX/EffectObject.writer.cs

```diff
@@ -140,7 +140,7 @@ public void Write(BinaryWriter writer, Options options)
 
         private static void WriteParameters(BinaryWriter writer, d3dx_parameter[] parameters, int count)
         {
-            writer.Write((byte)count);
+            writer.Write((short) count);
```

Comment:
How about ` writer.Write7BitEncodedInt(count)`?

Reply:
Yeah, I'll change that. It might save a couple bytes since most of the time count will fit in one byte. Thanks for the heads up, I didn't know about that function :)